src/Servers/Kestrel/shared/test/ServerRetryHelper.cs

```diff
@@ -34,23 +34,23 @@ public static async Task BindPortsWithRetry(Func<int, Task> retryFunc, ILogger l
             catch (Exception ex)
             {
                 retryCount++;
-                nextPortAttempt = port + 1;
+                nextPortAttempt = port + Random.Shared.Next(1000);
```

Comment:
```suggestion
                nextPortAttempt = port + Random.Shared.Next(100);
```
nit: Make it less likely that someone changes the retry count to be higher and exceed 65k port range limit.